src/core/lib/surface/call.cc

```diff
@@ -1178,12 +1178,6 @@ static void post_batch_completion(batch_control* bctl) {
         &call->metadata_batch[0 /* is_receiving */][0 /* is_trailing */]);
   }
   if (bctl->op.send_message) {
-    if (bctl->op.payload->send_message.stream_write_closed &&
-        error == GRPC_ERROR_NONE) {
-      error = grpc_error_add_child(
-          error, GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                     "Attempt to send message after stream was closed."));
```

Comment:
Why is it okay to change this behavior?

Reply:
We aren't really changing the behavior. Earlier, we were using the stream_write_closed flag to determine whether the writes were closed. After this, we would be using GRPC_ERROR_EOS